app/data_model/list_store.py

```diff
@@ -0,0 +1,82 @@
+import random
+from string import ascii_letters
+from collections import OrderedDict
+
+from structlog import get_logger
+
+from app.settings import EQ_LIST_ITEM_ID_LENGTH
+
+logger = get_logger()
+
+
+def random_string(length):
+    return ''.join(random.choice(ascii_letters) for _ in range(length))
+
+
+class ListStore:
+    """ Storage for lists.
+    """
+
+    def __init__(self, existing_items=None):
+        if not existing_items:
+            existing_items = []
+
+        self._lists = OrderedDict(existing_items)
+
+    def __getitem__(self, item):
+        try:
+            return self._lists[item]
+        except KeyError:
+            return list()
+
+    def _generate_identifier(self):
+        """ Generate an unused random 6 character string"""
+        while True:
+            candidate = random_string(EQ_LIST_ITEM_ID_LENGTH)
+            if candidate not in self.list_item_ids():
+                return candidate
+
+    def list_item_ids(self):
+        ids = []
+        for named_list in self._lists.values():
```

Comment:
can we not just `return self._lists.values()`

Reply:
We could. I don't think I would expect this to return multiple lists though.

Maybe we'd be better served by a `test_membership` method or similar instead.

Reply:
Ah yes, because its extending